src/test/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParserTest.java

```diff
@@ -266,4 +266,16 @@ public void testSubQuery() {
       Collections.singletonList(new SObjectDescriptor("Contacts", fields)),
       result.getChildSObjects());
   }
+
+  @Test
+  public void testKeywordInQuery() {
+    String query = "SELECT Data FROM BackgroundOperationResult";
+    SalesforceQueryParser.getObjectDescriptorFromQuery(query);
+  }
+
+  @Test
+  public void testEndKeywordInQuery() {
+    String query = "SELECT END FROM BackgroundOperationResult";
+    SalesforceQueryParser.getObjectDescriptorFromQuery(query);
+  }
```

Comment:
Can we also add a test for SCOPE keyword

Reply:
Added. 1ff5de85beff263ec032bc174f18f5fb929cf292